opencv_giflib.cpp

```diff
@@ -179,12 +157,39 @@ bool giflib_decoder_decode(giflib_decoder d, int frame_index, opencv_mat mat) {
         int prev_disposal = prevGCB.DisposalMode;
         if (prev_disposal == DISPOSE_BACKGROUND) {
             // draw over the previous frame with the BG color
-            // TODO should we do bounds checking here?
             SavedImage prevIM = d->gif->SavedImages[previous_frame_index];
             int prev_frame_left = prevIM.ImageDesc.Left;
             int prev_frame_top = prevIM.ImageDesc.Top;
             int prev_frame_width = prevIM.ImageDesc.Width;
             int prev_frame_height = prevIM.ImageDesc.Height;
+
+            if (prev_frame_left) {
```

Comment:
note: the next bit of code here is doing what's called a frame disposal. it's drawing over the section occupied by the previous frame with the gif's background color. this part calculates the skip dimensions slightly differently because it doesnt actually need to move the raster bits pointer. all it needs to do is not try to draw outside the bounds of the frame.